### PR TITLE
Update mysql integration

### PR DESCRIPTION
**Pull request by jsoriano**
## What does this PR do?

* Sync mysql integration with beats, add test files.
* Add reference to ECS fields.
* Add dataset and module fields.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by jsoriano** [comment]
https://github.com/elastic/elastic-package/pull/396 is needed to handle the dropped events in pipeline tests.

File: packages/mysql/data_stream/slowlog/elasticsearch/ingest_pipeline/default.json [hunk 1]
```diff
@@ -1,10 +1,19 @@
 {
   "description": "Pipeline for parsing MySQL slow logs.",
   "processors": [{
+    "drop": {
+      "if": "ctx.message ==~ /^# Time:.*/ || ctx.message ==~ /^[\\/\\w\\.]+, Version: .*\\. started with:.*/ms"
```
Comment (mtojek) [inline packages/mysql/data_stream/slowlog/elasticsearch/ingest_pipeline/default.json:5]: nit: not just any comment (`# Something`)?

File: packages/mysql/data_stream/slowlog/manifest.yml [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
-        title: Slowlog paths
+        title: Paths
         multi: true
         required: true
         show_user: true
         default:
           - /var/log/mysql/mysql-slow.log*
-    title: MySQL slowlog logs
-    description: Collect MySQL slowlog logs
+          - /var/lib/mysql/{{.builtin.hostname}}-slow.log
```
Comment (mtojek) [inline packages/mysql/data_stream/slowlog/manifest.yml:15]: nit: did you check that this is supported correctly (`{{.builtin.hostname}}`)?

**Review by mtojek — APPROVED** [review]
LGTM unless the CI complains.

File: packages/mysql/data_stream/slowlog/elasticsearch/ingest_pipeline/default.json (hunk 1 above)
Comment (jsoriano) [reply packages/mysql/data_stream/slowlog/elasticsearch/ingest_pipeline/default.json:5]: Most of lines in these logs start with `#`, we only drop this one. I think that this time is redundant with other timestamp that is parsed in this module.

File: packages/mysql/data_stream/slowlog/manifest.yml (hunk 2 above)
Comment (jsoriano) [reply packages/mysql/data_stream/slowlog/manifest.yml:15]: Umm, I didn't see this one after reimporting, I will give it a try, good catch.

Comment (jsoriano) [reply packages/mysql/data_stream/slowlog/manifest.yml]: Trying this I found some issues and ended up adding the docker scenario and system tests :sweat_smile: (WIP, they don't pass yet).

But I am not still sure if this is supported, in any case I am going to use `/var/lib/mysql/*-slow.log*`, that should work the same for most if not all the cases, and we don't need to support these placeholders. This is the only module using them.

**Comment by jsoriano** [comment]
Moving back to draft till I check what happens with system tests.

**Review by mtojek — APPROVED** [review]
LGTM

Big plus for adding system tests. Nice!

**Comment by jsoriano** [comment]
System tests seem to be flaky, taking a look...

**Comment by jsoriano** [comment]
Ok, tests should be less flaky now. Things I have changed for that:
* Permissions for log files are managed explicitly in the entrypoint. Not sure if this is the best way to do it, but it is reliable.
* Added a service to wait for mysql to be healthy, this should avoid the errors collecting metrics.

@mtojek as follow ups I think that we should explore a couple of things:
* It seems that the filebeat registry persists between tests, not sure if it is because of this, but I have seen that when running the tests locally, eventually no logs are collected and I have to recreate the stack. It happens in this case with the error data stream but not with the slowlog one, the error one always collects the same file, while the slowlog one depends on the hostname. This makes me think that we should clean the registry between tests.
* For metrics, we are always want to be sure that the service is healthy, if not errors may happen, what make the tests fail. I think this should be managed by the service deployer, if not we are going to need to add a dummy service to all the docker compose files.

**Comment by mtojek** [comment]
Thanks for investigation. Both issues are good candidates for improvement. Feel free to create issues in elastic-package and we can discuss them there.

> It seems that the filebeat registry persists between tests, not sure if it is because of this, but I have seen that when running the tests locally, eventually no logs are collected and I have to recreate the stack. It happens in this case with the error data stream but not with the slowlog one, the error one always collects the same file, while the slowlog one depends on the hostname. This makes me think that we should clean the registry between tests.

This one is mysterious to me as different test files (different filenames) are used. We have never observed issues with filebeat registry. I always suggest to depend on live logs instead of copied files. 
I know that there might be a problem when you recreate the `~/.elastic-package/tmp/service_logs` directory as the mounted path is lost and the agent can't see new location. I'm fine with investigating it further.

File: packages/mysql/_dev/deploy/docker/entrypoint.sh [hunk 3]
```diff
@@ -1,6 +1,14 @@
 #!/bin/bash
 
+# mysqld creates log files without read permissions for others,
```
Comment (mtojek) [inline packages/mysql/_dev/deploy/docker/entrypoint.sh:3]: Nice hack!

**Review by mtojek — APPROVED** [review]

**Comment by jsoriano** [comment]
Follow up issues created: https://github.com/elastic/elastic-package/issues/405, https://github.com/elastic/elastic-package/issues/406.

**Pull request merged**